spec/integration/system_pager_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+RSpec.describe TTY::Pager::SystemPager do
+  around :each do |example|
+    Dir.mktmpdir do |dir|
+      FileUtils.cd(dir) do
+        example.run
+      end
+    end
+  end
+
+  it "paginates some text" do
+    output_path = Pathname.new("output.txt")
+    described_class.page("Some text", command: "ruby #{fixtures_path("cat.rb")} > #{output_path}")
+
+    expect(output_path.read).to eq "Some text"
+  end
+
+  it "allows pagination to happen asynchronously" do
+    output_path = Pathname.new("output.txt")
+    pager_command = "ruby #{fixtures_path("external_pager.rb")} #{output_path}"
+
+    described_class.page(command: pager_command) do |pager|
+      pager.puts("one")
+      pager.write("two\n")
+      pager.try_write("three\n")
+    end
+
+    expect(output_path.read).to eq "one\ntwo\nthree\n"
+  end
+
+  it "stops paginating once external tool is closed" do
+    output_path = Pathname.new("output.txt")
+    pager_command = "ruby #{fixtures_path("external_pager.rb")} #{output_path}"
+
+    described_class.page(command: pager_command) do |pager|
+      pager.puts("one")
+      pager.puts("two")
+
+      # wait for script to finish writing
+      sleep 0.5
+
+      pid = pager.instance_variable_get("@pager_io").instance_variable_get("@pid")
+      Process.kill("KILL", pid)
+      Process.waitpid(pid)
```

Comment:
Newer Ruby on Windows raises broken pipe error (see https://ci.appveyor.com/project/piotrmurach/tty-pager/builds/33096082/job/y9yg2821nxbgobh7#L86). Maybe we could wrap this in begin-rescue?

Reply:
Hm, I thought it might have been this: https://github.com/piotrmurach/tty-pager/pull/24/commits/5a41260ab69ea99389809811790cf160f8033fc5

And, actually, the current build's specs seem to have no failures, but they still exit with an error code? https://ci.appveyor.com/project/piotrmurach/tty-pager/builds/33097333/job/qs91jr23mvab8gns

I'm not sure what the deal is there. I can try wrapping the test, but I don't understand why the error doesn't happen in that commit. I'll experiment later today.

Reply:
I experimented on Windows and may have found a 'clean' way to simulate pager failure without a need to kill a process via api. I suggest we add a new `broken_pager.rb` fixture file with the following:

```ruby
ARGF.read.each_line do |line|
  if line.match(/KILL_PROCESS/)
    exit(1)
  end
  STDOUT.puts(line)
end
```
Then the test can be rewritten to:

```ruby
TTY::Pager::SystemPager.page(command: "ruby broken_pager.rb > output.txt") do |pager|
  pager.puts("one")
  pager.puts("two")
  pager.puts("KILL_PROCESS")
  pager.puts("three")
end

expect(output_path.read).to eq "one\ntwo\n"
```

I double-checked and this test for sure raises no child process error if the rescue close in the `PagerIO`  is commented out.

If you're happy with this solution I can squash/merge your PR and resolve this issue?

Reply:
Yeah, this seems reasonable to me :+1: